core/trino-main/src/test/java/io/trino/operator/TestDistinctLimitOperator.java

```diff
@@ -73,119 +73,121 @@ public void tearDown()
         scheduledExecutor.shutdownNow();
     }
 
-    @DataProvider
-    public Object[][] dataType()
+    @Test
+    public void testDistinctLimit()
     {
-        return new Object[][] {{VARCHAR}, {BIGINT}};
+        for (boolean hashEnabled : Arrays.asList(true, false)) {
+            DriverContext driverContext = newDriverContext();
+
+            RowPagesBuilder rowPagesBuilder = rowPagesBuilder(hashEnabled, Ints.asList(0), BIGINT);
+            List<Page> input = rowPagesBuilder
+                    .addSequencePage(3, 1)
+                    .addSequencePage(5, 2)
+                    .build();
+
+            OperatorFactory operatorFactory = new DistinctLimitOperator.DistinctLimitOperatorFactory(
+                    0,
+                    new PlanNodeId("test"),
+                    rowPagesBuilder.getTypes(),
+                    Ints.asList(0),
+                    5,
+                    rowPagesBuilder.getHashChannel(),
+                    joinCompiler);
+
+            MaterializedResult expected = resultBuilder(driverContext.getSession(), BIGINT)
+                    .row(1L)
+                    .row(2L)
+                    .row(3L)
+                    .row(4L)
+                    .row(5L)
+                    .build();
+
+            assertOperatorEquals(operatorFactory, driverContext, input, expected, hashEnabled, ImmutableList.of(1));
+        }
     }
 
-    @DataProvider(name = "hashEnabledValues")
-    public static Object[][] hashEnabledValuesProvider()
+    @Test
+    public void testDistinctLimitWithPageAlignment()
     {
-        return new Object[][] {{true}, {false}};
+        for (boolean hashEnabled : Arrays.asList(true, false)) {
+            DriverContext driverContext = newDriverContext();
+
+            RowPagesBuilder rowPagesBuilder = rowPagesBuilder(hashEnabled, Ints.asList(0), BIGINT);
+            List<Page> input = rowPagesBuilder
+                    .addSequencePage(3, 1)
+                    .addSequencePage(3, 2)
+                    .build();
+
+            OperatorFactory operatorFactory = new DistinctLimitOperator.DistinctLimitOperatorFactory(
+                    0,
+                    new PlanNodeId("test"),
+                    rowPagesBuilder.getTypes(),
+                    Ints.asList(0),
+                    3,
+                    rowPagesBuilder.getHashChannel(),
+                    joinCompiler);
+
+            MaterializedResult expected = resultBuilder(driverContext.getSession(), BIGINT)
+                    .row(1L)
+                    .row(2L)
+                    .row(3L)
+                    .build();
+
+            assertOperatorEquals(operatorFactory, driverContext, input, expected, hashEnabled, ImmutableList.of(1));
+        }
     }
 
-    @Test(dataProvider = "hashEnabledValues")
-    public void testDistinctLimit(boolean hashEnabled)
```

Comment:
Rather than removing these methods, which also disturbs the indentation and makes the changes larger than necessary, keep the methods and call them from the test method:
```java
@Test
public void testDistinctLimit()
{
    testDistinctLimit(true);
    testDistinctLimit(false);
}
```
This is easier to read than a for loop and makes it easier to debug a specific test case.